### PR TITLE
Unit Test: QBO Components

### PR DESCRIPTION
**Pull request by anishfyle**
Screenshot 2024-09-03 at 6 40 45 PM

Screenshot 2024-09-03 at 9 37 33 PM

Screenshot 2024-09-03 at 9 37 45 PM

### CODE REVIEW

File: src/app/integrations/qbo/qbo.component.spec.ts [hunk 1]
```diff
@@ -1,23 +1,111 @@
...
   let fixture: ComponentFixture<QboComponent>;
+  let helperServiceSpy: jasmine.SpyObj<HelperService>;
+  let qboHelperServiceSpy: jasmine.SpyObj<QboHelperService>;
+  let routerSpy: jasmine.SpyObj<Router>;
+  let storageServiceSpy: jasmine.SpyObj<StorageService>;
+  let userServiceSpy: jasmine.SpyObj<IntegrationsUserService>;
+  let workspaceServiceSpy: jasmine.SpyObj<WorkspaceService>;
+  let windowServiceMock: Partial<WindowService>;
+
+  const mockUser: MinimalUser = {
```
Comment (DhaaraniCIT) [inline src/app/integrations/qbo/qbo.component.spec.ts:25]: can create fixture and call the values, so the file will be clean

File: src/app/integrations/qbo/qbo-onboarding/qbo-onboarding-landing/qbo-onboarding-landing.component.spec.ts [hunk 2]
```diff
@@ -20,4 +49,14 @@ xdescribe('QboOnboardingLandingComponent', () => {
...
+  it('should call oauthHandler when connectQbo is called', () => {
+    component.connectQbo();
+    expect(helperServiceSpy.oauthHandler).toHaveBeenCalled();
+  });
+
+  it('should set qboConnectionInProgress to true when connectQbo is called', () => {
+    component.connectQbo();
+    expect(component.qboConnectionInProgress).toBeTrue();
+  });
+});
```
Comment (ashwin1111) [inline src/app/integrations/qbo/qbo-onboarding/qbo-onboarding-landing/qbo-onboarding-landing.component.spec.ts:62]: cover 100% of lines

File: src/app/integrations/qbo/qbo-shared/qbo-employee-settings/qbo-employee-settings.component.spec.ts [hunk 3]
```diff
@@ -20,4 +56,14 @@ xdescribe('QboEmployeeSettingsComponent', () => {
...
+
+  it('should setup form on init', fakeAsync(() => {
+    component.ngOnInit();
+    tick();
+    fixture.detectChanges();
+    expect(component.employeeSettingForm).toBeDefined();
+    expect(component.employeeSettingForm.controls.employeeMapping).toBeDefined();
+    expect(component.employeeSettingForm.controls.autoMapEmployee).toBeDefined();
+    expect(component.employeeSettingForm.controls.searchOption).toBeDefined();
+  }));
```
Comment (ashwin1111) [inline src/app/integrations/qbo/qbo-shared/qbo-employee-settings/qbo-employee-settings.component.spec.ts:68]: need to cover all lines and cases

File: src/app/integrations/qbo/qbo.fixture.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    access_token: 'mock_access_token',
+    refresh_token: 'mock_refresh_token',
+    full_name: 'Test User',
+    user_id: 'user123',
+    org_name: 'Test Org'
+};
+
+export const mockWorkspace = {
+    id: '1',
+    onboarding_state: QBOOnboardingState.CONNECTION
```
Comment (ashwin1111) [inline src/app/integrations/qbo/qbo.fixture.ts:16]: provide complete data with all keys

**Review by ashwin1111 — CHANGES_REQUESTED** [review]

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**